File: Ontology/DataSomeValuesFrom.cs
namespace Ontology
{
    public class DataSomeValuesFrom:
        DataPropertyRestriction,
        IDataSomeValuesFrom
    {
        private readonly IDataRange _dataRange;

        public DataSomeValuesFrom(
            IDataPropertyExpression dataPropertyExpression,
            IDataRange              dataRange
            ) : base(dataPropertyExpression)
        {
            _dataRange = dataRange;
        }

        IDataRange IDataSomeValuesFrom.DataRange => _dataRange;

        public override void Accept(
            IClassExpressionVisitor visitor
            ) => visitor.Visit(this);

        public override bool Evaluate(
            IClassMembershipEvaluator evaluator,
            object                    individual
            ) => evaluator.Evaluate(
                this,
                individual);
    }
}

File: Web/ClientApp/src/app/Ontology/DataSomeValuesFrom.ts
import { DataPropertyRestriction } from "./DataPropertyRestriction";
import { IClassExpressionSelector } from './IClassExpressionSelector';
import { IClassExpressionVisitor } from "./IClassExpressionVisitor";
import { IDataRange } from "./IDataRange";
import { IDataSomeValuesFrom } from "./IDataSomeValuesFrom";
import { IDataPropertyExpression } from "./IPropertyExpression";

export class DataSomeValuesFrom
    extends DataPropertyRestriction
    implements IDataSomeValuesFrom
{
    public constructor(
        dataPropertyExpression: IDataPropertyExpression,
        public DataRange      : IDataRange
        )
    {
        super(dataPropertyExpression)
    }

    Accept(
        visitor: IClassExpressionVisitor
        )
    {
        visitor.DataSomeValuesFrom(this);
    }

    Select<TResult>(
        selector: IClassExpressionSelector<TResult>
        ): TResult
    {
        return selector.DataSomeValuesFrom(this);
    }
}
